tests/ArraySchemaValidatorTest.php

```diff
@@ -31,40 +31,40 @@ public function testArrayOfStringsValid(): void {
 
     expect($validator->isValid())->toBeTrue();
     $validated = $validator->getValidatedInput();
-    expect(C\count($validated['array_of_strings'] ?? []))->toBeSame(4);
+    expect(C\count($validated['array_of_strings'] ?? vec[]))->toBeSame(4);
   }
 
   public function testArrayOfStringsLegacyArrays(): void {
-    $validator = new ArraySchemaValidator(['array_of_strings' => ['test', 'list', 'of', 'strings']]);
+    $validator = new ArraySchemaValidator(darray['array_of_strings' => varray['test', 'list', 'of', 'strings']]);
     $validator->validate();
 
     expect($validator->isValid())->toBeTrue();
     $validated = $validator->getValidatedInput();
-    expect(C\count($validated['array_of_strings'] ?? []))->toBeSame(4);
+    expect(C\count($validated['array_of_strings'] ?? vec[]))->toBeSame(4);
   }
 
   public function testArrayOfStringsLegacyAndHackArrays(): void {
     $validator = new ArraySchemaValidator(dict[
-      'array_of_strings' => ['test', 'list', 'of', 'strings'],
+      'array_of_strings' => varray['test', 'list', 'of', 'strings'],
     ]);
     $validator->validate();
 
     expect($validator->isValid())->toBeTrue();
     $validated = $validator->getValidatedInput();
-    expect(C\count($validated['array_of_strings'] ?? []))->toBeSame(4);
+    expect(C\count($validated['array_of_strings'] ?? vec[]))->toBeSame(4);
   }
 
   public function testUntypedArrayValid(): void {
     $validator = new ArraySchemaValidator(dict[
-      'untyped_array' => ['test', 'values'],
+      'untyped_array' => varray['test', 'values'],
     ]);
     $validator->validate();
 
     expect($validator->isValid())->toBeTrue();
     $validated = $validator->getValidatedInput();
 
-    expect(($validated['untyped_array'] ?? [])[0])->toBeSame('test');
-    expect(($validated['untyped_array'] ?? [])[1])->toBeSame('values');
+    expect(Shapes::idx($validated, 'untyped_array') as nonnull[0])->toBeSame('test');
```

Comment:
This is not strictly an array problem.
We are effectively testing that the untyped_array still has the same keys.
Saying `$validated['untyped_array'] ?? []` and accessing key 0 can fail in two ways.
Either the key `untyped_array` itself has vanished (defaulting to the []) or the array does not have a 0th index.
HHVM has also changed it behavior to throw on missing arraykeys instead, so `($nullable_array ?? [])[0]` will throw if $nullable_array ends up being null (`[]` does not have a zeroth key).
I tried using `Shapes::at()` to make these two failures different, nut that is a HHVM 4.6.0 feature :whoops:
The non disruptive change would have been to use `?? varray[]`, so if this is disliked, we can revert to that.